pmd-core/src/main/java/net/sourceforge/pmd/lang/LanguageRegistry.java

```diff
@@ -61,17 +57,8 @@ public static LanguageRegistry getInstance() {
         return instance;
     }
 
-    public static Collection<Language> getLanguages() {
-        // Filter out languages, that are not fully supported by PMD yet.
-        // Those languages should not have a LanguageModule then, but they have it.
-        // TODO This is unnecessary, if the incomplete language modules have been removed.
-        List<Language> languages = new ArrayList<>();
-        for (Language language : getInstance().languages.values()) {
-            if (language.getRuleChainVisitorClass() != null) {
-                languages.add(language);
-            }
-        }
-        return languages;
+    public static Set<Language> getLanguages() {
+        return new LinkedHashSet<>(getInstance().languages.values());
```

Comment:
maybe better use `Collections.unmodifiableSet` instead of a new `LinkedHashSet `?